### PR TITLE
[stable/sentry] resource limitations and file upload fix

### PR DESCRIPTION
**Pull request by ahmadalli**
Signed-off-by: ahmadali shafiee <mail@ahmadalli.net>

### CODE REVIEW

**Comment by ahmadalli** [comment]
the latest commit fixes https://github.com/getsentry/sentry/issues/5360

**Comment by maorfr** [comment]
/unassign @prydonius 
/unassign @lachie83 
/unassign @paulczar 
/unassign

/assign @okgolove 

Assigning the new chart maintainer for sentry! ;)

**Comment by ahmadalli** [comment]
@maorfr thanks :)

File: stable/sentry/values.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ web:
   replicacount: 1
   resources:
     limits:
-      cpu: 500m
+      cpu: 4000m
```
Comment (okgolove) [inline stable/sentry/values.yaml:17]: It is too big limit, isn't it?

File: stable/sentry/values.yaml [hunk 2]
```diff
@@ -58,7 +58,7 @@ worker:
   resources:
     limits:
       cpu: 300m
-      memory: 500Mi
+      memory: 8196Mi
```
Comment (okgolove) [inline stable/sentry/values.yaml]: The same idea, why 8Gb?

File: stable/sentry/templates/workers-deployment.yaml [hunk 3]
```diff
@@ -113,9 +113,18 @@ spec:
         - mountPath: /etc/sentry
           name: config
           readOnly: true
+        - mountPath: {{ .Values.persistence.filestore_dir }}
```
Comment (okgolove) [inline stable/sentry/templates/workers-deployment.yaml]: Let's follow CamelCase practice.

File: stable/sentry/Chart.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Sentry is a cross-platform crash reporting and aggregation platform.
 name: sentry
-version: 1.3.5
+version: 1.3.6
```
Comment (okgolove) [inline stable/sentry/Chart.yaml]: 1.4.0

**Review by okgolove — CHANGES_REQUESTED** [review]

File: stable/sentry/values.yaml (hunk 1 above)
Comment (ahmadalli) [reply stable/sentry/values.yaml:17]: as sentry [stated](https://docs.sentry.io/server/installation/), sentry is not designed to run on low resource systems. we faced too many container crashes on our productions systems with the previous limit which was resolved by increasing the resource limits

File: stable/sentry/values.yaml (hunk 2 above)
Comment (ahmadalli) [reply stable/sentry/values.yaml]: the same as above :)

File: stable/sentry/templates/workers-deployment.yaml (hunk 3 above)
Comment (ahmadalli) [reply stable/sentry/templates/workers-deployment.yaml]: I didn't change the name. (it was added on initial sentry commit https://github.com/helm/charts/commit/cb0060835763bcdd68fb080ab4799f36a1033fa1) I just added the same volume to the worker deployments

File: stable/sentry/values.yaml (hunk 1 above)
Comment (okgolove) [reply stable/sentry/values.yaml:17]: I'm sorry you faced the crashes.
But your requirements pushes user to have at least 2 x c5.2xlarge (for worker and web).
To be honest I'd like not to specify default resources.

Comment (ahmadalli) [reply stable/sentry/values.yaml:17]: you mean it's ok to remove them completely? or just keep the old limitations?
also does setting `limits` (and not `requests`) makes user to have the amount of resources that are stated on `limites` section?

File: stable/sentry/Chart.yaml (hunk 4 above)
Comment (ahmadalli) [reply stable/sentry/Chart.yaml]: I've bumped the version

File: stable/sentry/templates/workers-deployment.yaml (hunk 3 above)
Comment (maorfr) [reply stable/sentry/templates/workers-deployment.yaml]: Always leave the code cleaner then how you found it? :)

**Comment by maorfr** [comment]
I'm with @okgolove on all comments :)
Following https://helm.sh/docs/chart_best_practices/ is important...

File: stable/sentry/templates/workers-deployment.yaml (hunk 3 above)
Comment (ahmadalli) [reply stable/sentry/templates/workers-deployment.yaml]: I'm on it :)

Comment (ahmadalli) [reply stable/sentry/templates/workers-deployment.yaml]: also the value need to be passed as `SENTRY_FILESTORE_DIR` env. I'm adding it too

File: stable/sentry/values.yaml (hunk 1 above)
Comment (okgolove) [reply stable/sentry/values.yaml:17]: It doesn't, but:
```When Pods use more memory than their limit, a process that is using the most amount of memory, inside one of the pod's containers, will be killed by the kernel```
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/resource-qos.md#qos-classes

I think we should save old values, I'll rework it later.

Comment (ahmadalli) [reply stable/sentry/values.yaml:17]: the point is sentry seems useless with that low amount of resources. why should we have them as default and ship a broken chart to the users?

File: stable/sentry/templates/workers-deployment.yaml (hunk 3 above)
Comment (ahmadalli) [reply stable/sentry/templates/workers-deployment.yaml]: is it ok to remove this value and hardcode it? I don't see any case that changing `SENTRY_FILESTORE_DIR` from `/var/lib/sentry/files` (which is inside the container) to other values does any help

File: stable/sentry/values.yaml (hunk 1 above)
Comment (okgolove) [reply stable/sentry/values.yaml:17]: I wouldn't say it's broken.
For example, I'm using it with default resources and it works properly.
https://imgur.com/a/hy9hY2C
https://imgur.com/a/AMFtHFW

But if you're totally sure we shouldn't use the limits were specified before then please remove them at all and set:
```
resources: {}
  # We usually recommend not to specify default resources and to leave this as a conscious
  # choice for the user. This also increases chances charts run on environments with little
  # resources, such as Minikube. If you do want to specify resources, uncomment the following
  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
  # limits:
  #  cpu: 100m
  #  memory: 128Mi
  # requests:
  #  cpu: 100m
  #  memory: 128Mi
```

File: stable/sentry/templates/workers-deployment.yaml (hunk 3 above)
Comment (okgolove) [reply stable/sentry/templates/workers-deployment.yaml]: I think in this case it's ok.

File: stable/sentry/values.yaml (hunk 1 above)
Comment (ahmadalli) [reply stable/sentry/values.yaml:17]: I've changed it to empty resources

**Review by okgolove — APPROVED** [review]

**Comment by okgolove** [comment]
@ahmadalli thank you!
Anything else?

**Comment by ahmadalli** [comment]
@okgolove thanks :)
I'll add another merge request for fixing #11879

**Comment by okgolove** [comment]
@ahmadalli could you rebase the branch?
Also, it seems the CI got stuck.

**Comment by maorfr** [comment]
/ok-to-test

Maybe this will help

**Comment by ahmadalli** [comment]
@okgolove I've merged master with current branch

**Comment by okgolove** [comment]
@ahmadalli set 1.4.0 please.

**Comment by okgolove** [comment]
@ahmadalli I'm so sorry but I have to ask to rebase the branch again, there was an another PR to the Sentry chart.

**Comment by ahmadalli** [comment]
@okgolove @maorfr I think I need to deploy this chart on a test env and see what causes the test to stuck

**Pull request closed without merging**